### PR TITLE
rtl-sdr: refactor, add rtl-sdr-blog and set it as default

### PR DESCRIPTION
**Pull request by Tungsten842**
## Description of changes

The current fork of rtl-sdr is quite outdated, it is [59 commits behind](https://github.com/librtlsdr/librtlsdr/compare/master...steve-m%3Alibrtlsdr%3Amaster) the osmoscom repo and the last update(released after 4 years), only added rtl-sdr v4 support and [some minimal fixes](https://github.com/librtlsdr/librtlsdr/commits/v0.9.0/).
This PR refactors the derivation to package different versions at the same time. The current fork is kept as it still includes some feature that are not present in the current version, the original osmocom version was already packaged and a new version made by rtlsdrblog is added. This fork is a frequently updated version directly based on the osmocom sources with some interesting [features](https://github.com/rtlsdrblog/rtl-sdr-blog) added, I have set this as the default version.
I have also added myself as a maintainer.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -34484,7 +34484,12 @@ with pkgs;
 
   rtl-ais = callPackage ../applications/radio/rtl-ais { };
 
-  rtl-sdr = callPackage ../applications/radio/rtl-sdr { };
+  inherit (callPackages ../applications/radio/rtl-sdr{})
```
Comment (bjornfor) [inline pkgs/top-level/all-packages.nix:34487]: There should be a space between the callPackage path argument and the empty attrset.

**Comment by Tungsten842** [comment]
Result of `nixpkgs-review pr 304063` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python312Packages.soapysdr-with-plugins</li>
  </ul>
</details>
<details>
  <summary>47 packages built:</summary>
  <ul>
    <li>abracadabra</li>
    <li>cubicsdr</li>
    <li>dabtools</li>
    <li>dump1090</li>
    <li>gnuradio</li>
    <li>gnuradio3_8</li>
    <li>gnuradio3_8Packages.ais</li>
    <li>gnuradio3_8Packages.osmosdr</li>
    <li>gnuradio3_8Packages.osmosdr.dev</li>
    <li>gnuradio3_9</li>
    <li>gnuradio3_9Packages.osmosdr</li>
    <li>gnuradio3_9Packages.osmosdr.dev</li>
    <li>gnuradioPackages.osmosdr</li>
    <li>gnuradioPackages.osmosdr.dev</li>
    <li>gqrx</li>
    <li>gqrx-gr-audio</li>
    <li>gqrx-portaudio</li>
    <li>guglielmo</li>
    <li>haskellPackages.rtlsdr</li>
    <li>haskellPackages.rtlsdr.doc</li>
    <li>kalibrate-rtl</li>
    <li>nrsc5</li>
    <li>openwebrx</li>
    <li>openwebrx.dist</li>
    <li>pothos</li>
    <li>python311Packages.pyrtlsdr</li>
    <li>python311Packages.pyrtlsdr.dist</li>
    <li>python311Packages.soapysdr-with-plugins</li>
    <li>python312Packages.pyrtlsdr</li>
    <li>python312Packages.pyrtlsdr.dist</li>
    <li>qradiolink</li>
    <li>rng-tools</li>
    <li>rtl-ais</li>
    <li>rtl-sdr (rtl-sdr-blog)</li>
    <li>rtl-sdr-librtlsdr</li>
    <li>rtl_433</li>
    <li>sdrangel</li>
    <li>sigdigger</li>
    <li>soapyrtlsdr</li>
    <li>soapysdr-with-plugins</li>
    <li>srsran</li>
    <li>suscan</li>
    <li>svxlink</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.librtlsdr</li>
    <li>urh</li>
    <li>urh.dist</li>
    <li>welle-io</li>
  </ul>
</details>

**Review by bjornfor — APPROVED** [review]

**Comment by markuskowa** [comment]
@GrahamcOfBorg build welle-io qradiolink nrsc5

**Review by markuskowa — APPROVED** [review]
I think I am okay with this. The rtl-sdr blog puts in a lot of effort into the topic, so it is probably the most up-to-date option.

**Pull request merged**